src/compiler/checker.ts

```diff
@@ -15712,6 +15712,21 @@ namespace ts {
             if (getModifierFlags(node) & ModifierFlags.Abstract && node.body) {
                 error(node, Diagnostics.Method_0_cannot_have_an_implementation_because_it_is_marked_abstract, declarationNameToString(node.name));
             }
+
+            // Is this the correct time to make assertions against the inheritance chain?

```

Comment:
This looks like the right place to do it - by the time we're here, we'll have resolved the base class